src/modules/land_detector/MulticopterLandDetector.cpp

```diff
@@ -131,115 +143,147 @@ void MulticopterLandDetector::_update_params()
-	if (!_actuator_armed.armed) {
+	if (!_armed) {
 		return true;
 	}
 
-	if (_has_minimal_thrust()) {
-		if (_min_trust_start == 0) {
-			_min_trust_start = hrt_absolute_time();
+
+	// minimal throttle: initially 10% of throttle range between min and hover
+	float sys_min_throttle = _params.minThrottle + (_params.hoverThrottle - _params.minThrottle) * 0.1f;
+
+	// Determine the system min throttle based on flight mode
+	if (!_flag_control_climb_rate_enabled) {
+		sys_min_throttle = (_params.minManThrottle + 0.01f);
+	}
+
+	// Check if thrust output is less than the minimum auto throttle param.
+	if (_actuator_controls_throttle <= sys_min_throttle) {
+		if (_min_thrust_start == 0) {
+			_min_thrust_start = hrt_absolute_time();
 		}
 
 	} else {
-		_min_trust_start = 0;
+		_min_thrust_start = 0;
+		return false;
 	}
 
-	float landThresholdFactor = 1.0f;
+
+	if (_freefall_hysteresis.get_state()) {
+		return false;
+	}
+
+
+	float landThresholdFactor = 1.f;
 
 	// Widen acceptance thresholds for landed state right after landed
 	if (hrt_elapsed_time(&_landed_time) < LAND_DETECTOR_LAND_PHASE_TIME_US) {
 		landThresholdFactor = 2.5f;
 	}
 
 	// Next look if all rotation angles are not moving.
-	float max_rotation_scaled = math::radians(_param_lndmc_rot_max.get()) * landThresholdFactor;
-
-	bool rotating = (fabsf(_vehicle_angular_velocity.xyz[0]) > max_rotation_scaled) ||
-			(fabsf(_vehicle_angular_velocity.xyz[1]) > max_rotation_scaled) ||
-			(fabsf(_vehicle_angular_velocity.xyz[2]) > max_rotation_scaled);
-
-	// Return status based on armed state and throttle if no position lock is available.
-	if (!_has_altitude_lock() && !rotating) {
-		// The system has minimum trust set (manual or in failsafe)
-		// if this persists for 8 seconds AND the drone is not
-		// falling consider it to be landed. This should even sustain
-		// quite acrobatic flight.
-		return (_min_trust_start > 0) && (hrt_elapsed_time(&_min_trust_start) > 8_s);
+	vehicle_angular_velocity_s vehicle_angular_velocity{};
+	_vehicle_angular_velocity_sub.copy(&vehicle_angular_velocity);
+	const Vector2f angular_velocity{vehicle_angular_velocity.xyz[0], vehicle_angular_velocity.xyz[1]};
+	const float max_rotation_scaled = math::radians(_param_lndmc_rot_max.get()) * landThresholdFactor;
+
+	if (angular_velocity.norm() > max_rotation_scaled) {
+		return false;
+	}
+
+	// If vertical velocity is available: ground contact, no thrust, no movement -> landed
+	if ((hrt_elapsed_time(&_vehicle_local_position.timestamp) < 1_s) && _vehicle_local_position.v_z_valid) {
+		return _ground_contact_hysteresis.get_state();
 	}
 
-	// Ground contact, no thrust and no movement -> landed
-	return _ground_contact_hysteresis.get_state() && _has_minimal_thrust() && !rotating;
+	// Otherwise, landed if the system has minimum thrust (manual or in failsafe) and no rotation for at least 8 seconds
+	return (_min_thrust_start > 0) && (hrt_elapsed_time(&_min_thrust_start) > 8_s);
```

Comment:
Summary of the big changes:
- low thrust threshold is mode/`control_climb_rate_enabled` dependent
- only rotation around body roll and pitch are considered, rotation around yaw gets ignored now
- freefall detection is considered